### DISCUSSION TITLE
Session timeout?

### DISCUSSION THREAD

**Discussion by lvojeroen** [Volue-Public/energy-mesh-python#192] [Q&A] [answered]
In my other question #189 I was using multiple parallel short-lived connections to exchange data with Mesh, re-creating a new session for every interaction with Mesh. This had a bad performance (~800 ms to read a time series) but after some discussions with your colleagues I understand the reason behind it. What I need to do, is to open one single session and use that to read/write data one-by-one. 

My question now is: is there a theoretical limit on how long a session can remain open? Does the session auto-close after a period of inactivity so do I need some kind of heartbeat query to keep it open?

**Comment by erny-powel** [comment, marked as answer]
> My question now is: is there a theoretical limit on how long a session can remain open? Does the session auto-close after a period of inactivity so do I need some kind of heartbeat query to keep it open?

I believe the current timeout is 10 hours without activity, but that number is going to become much lower for pure gRPC sessions. I definitely recommend planning for an active heartbeat if you want to have sessions that are inactive for more than a couple of minutes.
